server/sonar-server/src/main/java/org/sonar/server/computation/issue/EffortAggregator.java

```diff
@@ -77,36 +82,38 @@ public void afterComponent(Component component) {
     Metric metric = metricRepository.getByKey(CoreMetrics.TECHNICAL_DEBT_KEY);
 
     // total value
-    measureRepository.add(component, metric, Measure.newMeasureBuilder().create(this.currentDebt.minutes));
+    measureRepository.add(component, metric, Measure.newMeasureBuilder().create(this.effort.maintainabilityEffort));
 
+    // TODO delete following lines when working on SONAR-7425
     // distribution by rule
-    for (Map.Entry<Integer, Long> entry : currentDebt.minutesByRuleId.entrySet()) {
+    for (Map.Entry<Integer, Long> entry : effort.minutesByRuleId.entrySet()) {
       int ruleId = entry.getKey();
       long ruleDebt = entry.getValue();
       // debt can't be zero.
       measureRepository.add(component, metric, Measure.newMeasureBuilder().forRule(ruleId).create(ruleDebt));
     }
 
-    this.currentDebt = null;
+    this.effort = null;
   }
 
-  private class Debt {
-    private long minutes = 0L;
+  private class Effort {
+    private long maintainabilityEffort = 0L;
     private final SumMap<Integer> minutesByRuleId = new SumMap<>();
 
     void add(DefaultIssue issue) {
-      Long issueMinutes = issue.debtInMinutes();
-      if (issueMinutes != null && issueMinutes != 0L) {
-        this.minutes += issueMinutes;
-
-        Rule rule = ruleRepository.getByKey(issue.ruleKey());
-        this.minutesByRuleId.add(rule.getId(), issueMinutes);
+      Long issueEffort = issue.debtInMinutes();
+      if (issueEffort != null && issueEffort != 0L) {
+        if (issue.type().equals(RuleType.CODE_SMELL)) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1066)